src/plugins/discourse/fetch.js

```diff
@@ -0,0 +1,212 @@
+// @flow
+
+/**
+ * Class for retrieving data from the Discourse API.
+ *
+ * The Discourse API implements the JSON endpoints for all functionality of the actual site.
+ * As such, it tends to return a lot of information that we don't care about (in contrast
+ * to a GraphQL API which would give us only what we ask for). As such, we implement a simple
+ * interface over it, which both abstracts over calling the API, and does some post-processing
+ * on the results to simplify it to data that is relevant for us.
+ */
+
+import stringify from "json-stable-stringify";
+import fetch from "isomorphic-fetch";
+
+export type UserId = number;
+export type PostId = number;
+export type TopicId = number;
+
+export type Topic = {|
+  +id: TopicId,
+  +title: string,
+  +timestampMs: number,
+  +authorUsername: string,
+|};
+
+export type Post = {|
+  +id: PostId,
+  +topicId: TopicId,
+  // Which number post this was within the topic (starts at 1)
+  +postNumber: number,
```

Comment:
Consider calling this `indexInTopic` or similar? It’s not clear from the
names why “`id`” and “`postNumber`” should be two separate fields. (If
so, also rename `replyToPostIndex` for consistency.)